katello-attach-subscription

```diff
@@ -171,26 +800,114 @@ def vdcupdate()
     auto_attach = false
     keep_virt_only = false
     virtual_host = nil
+    desired_quantity = 1
+    sys_socket = 1
+    hypervisor_found = false
+    system_type = ""
+    this_system_cluster = ""
+    skip_hypervisor = false
+    sys = nil
+
+    # add to array system the name of the system itself taken from id
+    if @options[:verbose]
+      puts " VERBOSE: Current system #{currentcount+1}/#{systemstotal}: #{system['name']} (#{system['id']})"
+    end
 
-    # get detail for each system
+    # get detail for the current system to be checked.
     if @options[:debug]
-      puts "DEBUG: detail of the current system to be checked:"
+      puts " DEBUG: detail of the current system to be checked:"
       p system
     end
-    sys = api.resource(@host_resource).call(:show, {:id => system['id'], :fields => 'full'})
-    # add to array system the name of the system itself taken from id
-    puts "Current system: #{system['name']} (#{system['id']})"
+    sys = @api.resource(:hosts).call(:show, {:id => system['id'].to_i, :fields => 'full'})
+    if sys.count <= 0
+      puts "  ERROR: System name '#{system['name']}' not found. Is cache in use? Skipping."
+      next
+    end
+    if @options[:debug]
+      puts "  DEBUG: '#{system['name']}' id #{system['id']} :hosts api data:"
+      p sys
+    end
+    system_details[system['name']]=sys
+    # check if the type requested match the host one
+    # fixme: complete the check with the new API version of find type of server and the desired sub type
+    # if type key in system detail is present then we are using satellite version < 6.2
+    if sys.has_key?("type")
```

Comment:
I think this whole `system_type` block should be an own function, as this will allow us to add tests for it much easier